src/core/connection.c

```diff
@@ -2003,6 +2029,38 @@ QuicConnStart(
         goto Exit;
     }
 
+    if (Connection->Settings.IsSet.EncryptionOffloadEnabled) {
+        CXPLAT_QEO_CONNECTION Offloads[] = {
+            {
+                CXPLAT_QEO_OPERATION_ADD,
+                CXPLAT_QEO_DIRECTION_TRANSMIT,
+                CXPLAT_QEO_DECRYPT_FAILURE_ACTION_DROP,
+                0,
+                0,
+                CXPLAT_QEO_CIPHER_TYPE_AEAD_AES_256_GCM,
+                8,
+                {0},
+            },
+            {
+                CXPLAT_QEO_OPERATION_ADD,
+                CXPLAT_QEO_DIRECTION_RECEIVE,
+                CXPLAT_QEO_DECRYPT_FAILURE_ACTION_DROP,
+                0,
+                0,
+                CXPLAT_QEO_CIPHER_TYPE_AEAD_AES_256_GCM,
+                0,
+                {0},
+            }
+        };
+        Status = CxPlatSocketUpdateQeo(Path->Binding->Socket, Offloads, 2);
+        if (QUIC_FAILED(Status)) {
+            // TODO: query QEO capability, set CXPLAT_DATAPATH_FEATURE_ENCRYPTION_OFFLOAD
+            goto Exit;
+        }
```

Comment:
Failure here shouldn't be fatal. Instead, only set the offloading flag to true on success.